internal/metamorph/store/postgresql/postgres.go

```diff
@@ -577,26 +578,43 @@ func (p *PostgreSQL) UpdateStatusBulk(ctx context.Context, updates []store.Updat
 		if update.Error != nil {
 			rejectReasons[i] = update.Error.Error()
 		}
+
+		var historyDataStr *string
+		if update.StatusHistory != nil {
+			historyData, err := prepareStructForSaving(update.StatusHistory)
+			if err != nil {
+				return nil, err
+			}
+			historyStr := string(historyData)
+			historyDataStr = &historyStr
+		}
+		statusHistories[i] = historyDataStr
 	}
 
 	qBulk := `
 		UPDATE metamorph.transactions
 			SET
-			status=bulk_query.status,
-			reject_reason=bulk_query.reject_reason,
-			last_modified=$1,
-			status_history=status_history || json_build_object(
-				'status', metamorph.transactions.status,
-				'timestamp', last_modified
-			)::JSONB
+				status = bulk_query.status,
+				reject_reason = bulk_query.reject_reason,
+				last_modified = $1,
+				status_history = status_history 
+					|| COALESCE(
+						json_build_object(
+							'status', metamorph.transactions.status,
+							'timestamp', last_modified
+						)::JSONB || bulk_query.history_update,
+						json_build_object(
+							'status', metamorph.transactions.status,
+							'timestamp', last_modified
+						)::JSONB
+					)
 			FROM
 			(
-				SELECT *
-						FROM
-						UNNEST($2::BYTEA[], $3::INT[], $4::TEXT[])
-				AS t(hash, status, reject_reason)
+				SELECT t.hash, t.status, t.reject_reason, h.history_update
+				FROM UNNEST($2::BYTEA[], $3::INT[], $4::TEXT[]) AS t(hash, status, reject_reason)
+				LEFT JOIN UNNEST($5::JSONB[]) AS h(history_update) ON TRUE
```

Comment:
Why LEFT JOIN here? Will this not work?
```suggestion
				SELECT t.hash, t.status, t.reject_reason, t.history_update
				FROM UNNEST($2::BYTEA[], $3::INT[], $4::TEXT[], $5::JSONB[]) AS t(hash, status, reject_reason, history_update)
```

Reply:
Yes it will work, I forgot to change it after trying many solutions. Now it should be ok